modules/core/client/app/init.js

```diff
@@ -36,11 +36,13 @@ angular.module(ApplicationConfiguration.applicationModuleName).run(function ($ro
         }
       }
     }
+
   });
 
   // Record previous state
   $rootScope.$on('$stateChangeSuccess', function (event, toState, toParams, fromState, fromParams) {
     storePreviousState(fromState, fromParams);
+    $rootScope.$state = $state;
```

Comment:
Is this still needed? Doesn't the `getWorkingState` function in your directive take care of getting the current state?